src/app/collective-rewards/settings/builder/BuilderRewardsSettingsForm.tsx

```diff
@@ -0,0 +1,120 @@
+import { Button } from '@/components/Button'
+import { Form, FormControl, FormField, FormInput, FormItem, FormLabel, FormMessage } from '@/components/Form'
+import { LoadingSpinner } from '@/components/LoadingSpinner'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { useRouter } from 'next/navigation'
+import { FC, useEffect, useState } from 'react'
+import { useForm } from 'react-hook-form'
+import { GoRocket } from 'react-icons/go'
+import { Address } from 'viem'
+import { useAccount } from 'wagmi'
+import { z } from 'zod'
+import { useGetBackerRewardsForBuilder, useSetBackerRewardsForBuilder } from '../hooks'
+import { percentageToWei } from '../utils'
+import { useBuilderSettingsContext } from './BuilderSettings'
+
+const formSchema = z.object({
+  reward: z
+    .string()
+    .regex(
+      /^\d{1,3}(\.\d{1,18})?$/,
+      'Invalid format: should be up to 3 digits before decimal and up to 18 digits after decimal.',
+    )
+    .refine(
+      value => {
+        const digitsOnly = value.replace('.', '')
+
+        if (digitsOnly.length > 19) {
+          return false
+        }
+
+        const [whole, decimal] = value.split('.')
+
+        // Check that if there are 3 digits before the decimal, they must be "100" and there are no non-zero digits after 100.
+        return whole.length < 3 || (whole === '100' && (!decimal || BigInt(decimal) == 0n))
+      },
+      {
+        message:
+          'Value must be between 0.000_000_000_000_000_1 and 100.0, with total number of digits (excluding decimal point) less or equal to 19.',
```

Comment:
Do we really want the users to see this? Let's discuss with product, but I'd consider up until 2 decimals. After all we're talking about percentage.

Reply:
Yea, but we're percentages of potentially quite large values. This will only show if they enter too many decimals, too many digits or a value over 100. The first two are not likely to happen that often, but if they do, then it's probably because the user needed the precision. The last one might only happen as a mistake.
I don't know. I'd keep it but, I'm happy to change it to whatever the consensus is .)